[fix][win32][depends] This should resolve the python issues caused by vs2015 upgrade.

Paxxi opened the pull request:
Fixed copying of dll during build to avoid risking using python27.dll from system32
Rebuilt all statically linked libs with _ITERATOR_DEBUG_LEVEL=0 because of performance issues, especially during video development
Curl should once again be linked against openssl
switched all projects to use /MD or /MDd because it enables sharing of env between the loaded modules making our environment hacks less needed.
Added curl and openssl to ignore list to avoid them being hooked by our crt emulation. This needs a proper fix and is likely resolved by Voyager's PR
@mapfau would love to see if this works for you, fully understand if you don't want to mess with your setup though
@afedchin can you test the performance in debug mode with this and see if it solves the issue you were seeing?

afedchin commented:
I'll test this today night

afedchin commented on xbmc/cores/DllLoader/Win32DllLoader.cpp:
```diff
@@ -33,7 +33,8 @@
 
-extern "C" FILE _iob[];
+//This shouldn't be needed anymore and it's not exported through the dll
+//extern "C" FILE _iob[];
```
this should go if it's not needed

Paxxi replied:
yep you're right on both

afedchin commented on xbmc/cores/DllLoader/Win32DllLoader.cpp:
```diff
@@ -126,7 +127,7 @@ Export win32_exports[] =
 
   // reading/writing from stdin/stdout needs this
-  { "_iob",                       -1, (void*)&_iob,                         NULL },
+  //{ "_iob",                       -1, (void*)&_iob,                         NULL },
```
same as above

Paxxi committed:
This should resolve the python issues caused by vs2015 upgrade.
Fixed copying of dll during build to avoid risking using python27.dll from system32
Rebuilt all statically linked libs with _ITERATOR_DEBUG_LEVEL=0 because of performance issues, especially during video development
Curl should once again be linked against openssl
switched all projects to use /MD or /MDd because it enables sharing of env between the loaded modules making our environment hacks less needed.
Added curl and openssl to ignore list to avoid them being hooked by our crt emulation. This needs a proper fix and is likely resolved by Voyager's PR

mapfau commented:
thanx @Paxxi for your hard work on it! I'm currently finalizing the smoothstream plugin and will directly test after this.

Paxxi commented:
@fetzerch can you check the cmake fixes? I've tested that it builds but some experienced eyes would be nice :) Also this supersedes #9552 so I'm closing it

Paxxi committed:
Remove commented stuff, update cmake builds as well and ofc, packaging crossguid failed so bump it to v3

Tolriq commented:
@Paxxi sorry to ask here but I do not see any official thread in forums.

Is the switch done and VS 2013 no more supported or is it still in preparation ?

I'm lost with all those PR + cmake and don't know what I should use for my next PRs :(

Paxxi commented:
@Tolriq correct, the switch is done and 2013 is no longer supported since last weekend.

afedchin commented:
@Paxxi I didn't see a difference. same code but compiled with vs2013 works perfectly.

afedchin commented:
@Paxxi in additional seems overall performance has increased slightly

Paxxi commented:
@afedchin probably because you still have the vs2013 libs still on disk, cleaning the repo or a new checkout will not download vs2013 compatible static libraries.

afedchin commented:
I'll try to delete all depends and download them again.

afedchin commented:
@Paxxi unfortunately, I get same results. actually I have solved main performance issue, but still strange that on vs2013 it was no issue.

Paxxi commented:
Yeah that is weird, need to investigate further and see what might be causing it

fetzerch commented:
@Paxxi: Changes to the CMake files look good :)

Paxxi commented:
jenkins build this please

Paxxi commented:
well let's get this in so nightlies have a chance at working again, any further issues can be sorted as we go along

kodibrasil commented:
@Paxxi, windows OS now have error load Libcurl.dll, look this post with debug log, file exist in the path, but have error load.
http://forum.kodi.tv/showthread.php?tid=268913&pid=2309968#pid2309968

Paxxi commented:
Thanks for the report, will try and investigate
